modules/users/server/controllers/users/users.authentication.server.controller.js

```diff
@@ -79,10 +79,32 @@ exports.signout = function(req, res) {
 };
 
 /**
+ * OAuth provider call
+ */
+exports.oauthCall = function(strategy, scope) {
+	return function(req, res, next) {
+		// Set redirection path on session.
+		// Do not redirect to a signin or signup page
+		if (!_.contains([
+			'/authentication/signin',
+			'/authentication/signup'
+		], req.query.redirect_to)) {
+			req.session.redirect_to = req.query.redirect_to;
+		}
+		// Authenticate
+		passport.authenticate(strategy, scope)(req, res, next);
+	};
+};
+
+/**
  * OAuth callback
  */
 exports.oauthCallback = function(strategy) {
 	return function(req, res, next) {
+		// Pop redirect URL from session
+		var sessionRedirectURL = req.session.redirect_to;
+		delete req.session.redirect_to;
```

Comment:
@ilanbiala thanks for looking into this. I am setting a property in the first call, `oauthCall`. In the callback I retrieve it. Do you think there would be a better way to implement this?